sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -453,40 +414,72 @@ private synchronized void resumeDistributeWorkflow() {
                 return;
             }
 
-            /* If we have a pending or notified download, check it. */
-            if (getStoredDownloadState() != DOWNLOAD_STATE_COMPLETED) {
-                if (mCheckedDownload) {
+            /* Load cached release details if process restarted and we have such a cache. */
+            int downloadState = getStoredDownloadState();
+            if (mReleaseDetails == null && downloadState != DOWNLOAD_STATE_COMPLETED) {
+                mReleaseDetails = DistributeUtils.loadCachedReleaseDetails();
```

Comment:
Do you have an expiration of the release details? What if it cached and newer release becomes available, looks it will get the cache until the cache gets removed.